collector/lib/ContainerMetadata.h

```diff
@@ -0,0 +1,44 @@
+#ifndef _CONTAINER_METADATA_H_
+#define _CONTAINER_METADATA_H_
+
+#include "system-inspector/EventExtractor.h"
+
+namespace collector {
+
+class ContainerMetadata {
+ public:
+  ContainerMetadata(sinsp* inspector) : inspector_(inspector) {
+    event_extractor_.Init(inspector);
+  }
+
+  inline std::string GetNamespace(sinsp_evt* event) {
+    const char* ns = event_extractor_.get_k8s_namespace(event);
+    return ns != nullptr ? ns : "";
+  }
+
+  inline std::string GetNamespace(const std::string& container_id) {
+    return GetContainerLabel(container_id, "io.kubernetes.pod.namespace");
+  }
+
+  inline std::string GetContainerLabel(const std::string& container_id, const std::string& label) {
+    const auto& containers = *inspector_->m_container_manager.get_containers();
```

Comment:
The Inspector service thread might be updating the container list while other threads call this through GetNamespace() (typically the instrospection API, or the filters)

Reply:
That's why I chose to use `get_containers()`, it returns a locked mutex:
https://github.com/stackrox/falcosecurity-libs/blob/fecec54dc75acde7c7ad186175a055efb1dff8fa/userspace/libsinsp/container.cpp#L286-L289